governance/program/src/state/proposal_instruction.rs

```diff
@@ -96,11 +99,15 @@ pub struct ProposalInstruction {
 
     /// Executed at flag
     pub executed_at: Option<UnixTimestamp>,
+
+    /// Instruction execution status
+    /// Note: The field is not used in the current version
+    pub execution_status: Option<InstructionExecutionStatus>,
```

Comment:
What will this be for?  Maybe you could have `InstructionExecutionStatus` as `NotRun`, `Success`, and `Error`, and then this doesn't need to be an `Option`

Reply:
I was debating with myself whether we should use `Option` to indicate `None` or a native enum value and it was a close call which you tipped in favour of enum value :).

I added the status to indicate whether the instruction was executed successfully or it returned an error. It means the default `None` status indicates it hasn't been attempted to run yet.
It is required to prevent a `Proposal` being permanently displayed as being in `Succeeded` or `Executing` state when one of its instruction is permanently broken and can't be executed successfully.
I left it as unused in this version because currently there is no clear way to determine whether instruction execution was successful because CPI call error always terminates parent transaction. 
I'm thinking the best option would be to make it optional in the runtime and make it possible to opt out of that behaviour and allow the parent to get CPI call errors.
However if that turns out to be impossible, for whatever reason, then the only option is to add a program instruction to manually flag instruction as broken.